exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -97,97 +96,75 @@
   public static final short MAX_VALIDATOR_COUNT_WITH_ENABLED_TIME_SERVICE = 3;
   @VisibleForTesting
   static final short MAX_SERVICE_NUMBER = 256;
+  @VisibleForTesting
+  static final int MAX_SERVICE_INSTANCE_ID = 1023;
   private static final Serializer<Block> BLOCK_SERIALIZER = BlockSerializer.INSTANCE;
-  private static final short TIME_SERVICE_ID = 4;
+  private final List<Integer> timeServiceIds;
 
-  private final Map<Short, Service> services = new HashMap<>();
   @VisibleForTesting
   final Cleaner snapshotCleaner = new Cleaner("TestKit#getSnapshot");
 
-  private TestKit(long nativeHandle, Map<Short, UserServiceAdapter> serviceAdapters) {
+  private TestKit(long nativeHandle, List<TimeServiceSpec> timeServiceSpecs) {
     super(nativeHandle, true);
-    populateServiceMap(serviceAdapters);
+    timeServiceIds = timeServiceSpecs.stream()
+        .map(t -> t.getServiceId)
+        .collect(toList());
   }
 
-  private static TestKit newInstance(List<Class<? extends ServiceModule>> serviceModules,
+  private static TestKit newInstance(TestKitServiceInstances[] serviceInstances,
                                      EmulatedNodeType nodeType, short validatorCount,
-                                     @Nullable TimeProvider timeProvider) {
-    Injector frameworkInjector = Guice.createInjector(new TestKitFrameworkModule());
-    return newInstanceWithInjector(serviceModules, nodeType, validatorCount, timeProvider,
-        frameworkInjector);
-  }
-
-  private static TestKit newInstanceWithInjector(
-      List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
-      short validatorCount, @Nullable TimeProvider timeProvider, Injector frameworkInjector) {
-    Map<Short, UserServiceAdapter> serviceAdapters = toUserServiceAdapters(
-        serviceModules, frameworkInjector);
+                                     List<TimeServiceSpec> timeServiceSpecs) {
```

Comment:
As we decided to instantiate `ServiceRuntime` at the Java side for Testkit, the arguments list should be extended with `ServiceRuntimeAdapter` instance.